synapse/storage/databases/main/events_worker.py

```diff
@@ -712,15 +712,19 @@ async def get_missing_events_from_db() -> Dict[str, EventCacheEntry]:
 
         return event_entry_map
 
-    async def _invalidate_get_event_cache(self, event_id: str) -> None:
-        # First we invalidate the asynchronous cache instance. This may include
-        # out-of-process caches such as Redis/memcache. Once complete we can
-        # invalidate any in memory cache. The ordering is important here to
-        # ensure we don't pull in any remote invalid value after we invalidate
-        # the in-memory cache.
+    # The next two methods implement get event cache invalidation the first invalidates
```

Comment:
three methods?

I'd much rather see this information in docstrings, or inside the methods: the methods really need docstrings anyway so it's easy to understand what they do without having to read a paragraph of text somewhere nearby.

Reply:
Much much nicer, bad practice on my part with the comment there! https://github.com/matrix-org/synapse/pull/13308/commits/d39d9bf11c9a713f56a4184b6f40d9d8907e7e5b